src/main/java/com/yahoo/identity/services/credential/CredentialServiceImplVulnerable.java

```diff
@@ -0,0 +1,43 @@
+package com.yahoo.identity.services.credential;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.yahoo.identity.services.key.KeyService;
+import com.yahoo.identity.services.key.KeyServiceImpl;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.BadRequestException;
+
+public class CredentialServiceImplVulnerable implements CredentialService {
+
+    private final KeyService keyService = new KeyServiceImpl();
+    private final Credential credential = new CredentialImpl();
+
+    @Override
+    @Nonnull
+    public Credential fromString(@Nonnull String credStr) {
+        try {
+
+            Algorithm algorithm = Algorithm.HMAC256(this.keyService.getSecret("cookie"));
+
+            JWTVerifier verifier = JWT.require(algorithm)
+                .acceptLeeway(1)   // 1 sec for nbf and iat
+                .acceptExpiresAt(5)   // 5 secs for exp
```

Comment:
Do we need to set those tolerance values? I think our timeout should be long enough to cover the time drift.

Reply:
No. We don't if the expiration time is long enough. Now removed.